components/ConvertSteps/Converter.js

```diff
@@ -0,0 +1,195 @@
+        setStepperStage('error')
+        console.log(err)
+      }
+    },
+    [claimOrder, applyStepState, toAnj]
+  )
+
+  const handleBuyOrderStep = useCallback(async () => {
+    const stepType = 'buyOrder'
+
+    setCurrentStep(stepType)
+
+    try {
+      // Awaiting confirmation
+      applyStepState(stepType, 'waiting')
+      const transaction = await openOrder(amountSource, toAnj)
+
+      // Mining transaction
+      applyStepState(stepType, 'working', transaction.hash)
+      await transaction.wait()
+
+      // Success
+      applyStepState(stepType, 'success')
+      handleClaimOrderStep(transaction.hash)
+    } catch (err) {
+      applyStepState(stepType, 'error')
+      setStepperStage('error')
+      console.log(err)
+    }
+  }, [amountSource, openOrder, applyStepState, toAnj, handleClaimOrderStep])
+
+  const handleApprovalStep = useCallback(async () => {
+    const stepType = 'approval'
+
+    setCurrentStep(stepType)
+
+    try {
+      // Awaiting approval
+      applyStepState(stepType, 'waiting')
+      await checkAllowance(amountSource)
+
+      // Success
+      applyStepState(stepType, 'success')
+      handleBuyOrderStep()
+    } catch (err) {
+      applyStepState(stepType, 'error')
+      setStepperStage('error')
+      console.log(err)
+    }
+  }, [amountSource, checkAllowance, applyStepState, handleBuyOrderStep])
+
+  function handleRepeatTransaction() {
+    if (currentStep === 'approval') {
+      handleApprovalStep()
+    } else if (currentStep === 'buyOrder') {
+      handleBuyOrderStep()
+    } else if (currentStep === 'claimOrder') {
+      handleClaimOrderStep(stepState.buyOrder.hash)
+    }
+  }
+
+  useEffect(() => {
+    if (amountSource) {
+      toAnj ? handleApprovalStep() : handleBuyOrderStep()
+    }
+  }, [amountSource, handleApprovalStep, handleBuyOrderStep, toAnj])
+
+  return (
+    <StepperLayout
+      fromAmount={formattedFromAmount}
+      toAmount={formattedToAmount}
+      stage={stepperStage}
+      toAnj={toAnj}
+      onRepeatTransaction={handleRepeatTransaction}
+      onReturnHome={handleReturnHome}
+    >
+      {toAnj && (
+        <>
+          <Step
+            title="Approve ANT"
+            number="1"
+            active={stepState.approval.active}
+            status={stepState.approval.status}
+          />
+          <Divider />
+        </>
+      )}
+
+      <Step
+        title="Create buy order"
+        number={toAnj ? '2' : '1'}
+        active={stepState.buyOrder.active}
+        status={stepState.buyOrder.status}
+        transactionHash={stepState.buyOrder.hash}
+      />
+      <Divider />
+      <Step
+        title="Claim order"
+        number={toAnj ? '3' : '2'}
+        active={stepState.claimOrder.active}
+        status={stepState.claimOrder.status}
+        transactionHash={stepState.claimOrder.hash}
+      />
+    </StepperLayout>
+  )
+}
+
+Converter.propTypes = {
+  toAnj: PropTypes.bool,
+  handleReturnHome: PropTypes.func,
```

Comment:
Nitpick: we'll generally name handlers given as props with `on*` and name internally defined handlers as `handle*`

Reply:
👍